hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/StorageType.java

```diff
@@ -33,13 +33,12 @@
 @InterfaceAudience.Public
 @InterfaceStability.Unstable
 public enum StorageType {
-  // sorted by the speed of the storage types, from fast to slow
   RAM_DISK(true, true),
-  NVDIMM(false, true),
   SSD(false, false),
   DISK(false, false),
   ARCHIVE(false, false),
-  PROVIDED(false, false);
+  PROVIDED(false, false),
+  NVDIMM(false, true);
 
```

Comment:
Because of addition of new STORAGE TYPE, 
during upgrade
calls from old clients, who doesnt have this storagetype, will experience failure.

Especially `getQuotaUsage()` call will experience failure. May be need to analyze more to avoid failures during upgrade.
Even though upgrade is finalized, old clients should continue to work.

Reply:
We add check for setQuota() in FSNamesystem.java, I think it's ok, right?

Reply:
I am not sure but will getStoragePolicies also land up in something similar issue? Due to unavailability of storage type? The quota stuff shall be there for PROVIDED also but in case this backward incompatibility is there with Storage Policy too, Then we need to find out some way.
@vinayakumarb do you have pointers or suggestions on this, how to tackle this?

Reply:
I have verified the `getStoragePolicies()` with older clients.
Older clients will get the storage policy but, unknown storage types will be ignored.
So in this case, ALLNVDIMM storage policy shows empty StorageTypes for older clients.

May be need to show DEFAULT StorageType instead of ignoring the unknown StorageTypes. This also can be fixed in a separate Jira. Right now, existing clients wont be broken on` getStoragePolicies()` call with this change.

So nothing special required for that in this PR.

Reply:
Thanks for clarification and testing.